GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -74,43 +79,83 @@ gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
   }
 };
 
-// This method is asynchronous and return a promise with the fontName.
-gdjs.BitmapFontManager.prototype.loadBitmapFont = async function (
+/**
+ * @param {string} bitmapFontResourceName
+ * @param {PIXI.Texture} texture
+ * @returns {PIXI.BitmapFont}
+ */
+gdjs.BitmapFontManager.prototype.getBitmapFontFromData = function (
   bitmapFontResourceName,
   texture
 ) {
+  // Reuse an existing bitmapFont that is already in memory and already installed
+  for (var i = 0; i < this._loadedFontsData.length; ++i) {
+    let loadedFontData = this._loadedFontsData[i];
+    if (
+      loadedFontData.fontResourceFileName === bitmapFontResourceName &&
+      PIXI.BitmapFont.available[loadedFontData.fontName]
+    ) {
+      return PIXI.BitmapFont.available[loadedFontData.fontName];
+    }
+  }
+
   let bitmapFontResource = this._resources.find(function (resource) {
     return (
       resource.kind === 'bitmapFont' && resource.name === bitmapFontResourceName
     );
   });
 
   if (!bitmapFontResource) {
-    return Promise.reject(
-      new Error(
-        'Can\'t find resource with name: "' +
-          bitmapFontResourceName +
-          '" (or is not a bitmapFont file).'
-      )
+    console.warn(
+      'Can\'t find resource with name: "' +
+        bitmapFontResourceName +
+        '" (or is not a bitmapFont file).'
     );
-  }
 
-  let response = await fetch(bitmapFontResource.file);
-  if (!response.ok) {
-    return Promise.reject(
-      new Error(
-        "Can't fetch the bitmap font file, error: " +
-          response.status +
-          '(' +
-          response.statusText +
-          ')'
-      )
-    );
+    if (!PIXI.BitmapFont.available['Arial-20-#ffffff-bitmapFont']) {
+      let bitmapFontStyle = new PIXI.TextStyle();
+      bitmapFontStyle.fontFamily = 'Arial';
+      bitmapFontStyle.fontSize = 20;
+      bitmapFontStyle.align = 'left';
+      bitmapFontStyle.wordWrap = true;
+      bitmapFontStyle.fill = '#ffffff';
+
+      const defaultSlugFontName =
+        bitmapFontStyle.fontFamily +
+        '-' +
+        bitmapFontStyle.fontSize +
+        '-' +
+        bitmapFontStyle.fill +
+        '-bitmapFont';
+
+      //Generate default bitmap font
+      PIXI.BitmapFont.from(defaultSlugFontName, bitmapFontStyle, {
+        chars: [
+          [' ', '~'], // All the printable ASCII characters
+        ],
+      });
+      PIXI.BitmapFont.available['Arial-20-#ffffff-bitmapFont'].font =
```

Comment:
This seems dangerous, you're modifying a PIXI private object. The documentation says that it's read only:
![image](https://user-images.githubusercontent.com/1280130/106533405-055c6c00-64ea-11eb-8c2d-993c52c126f7.png)

So you probably don't want to add a "font" property on a BitmapFont. This can create very hard to debug issues and decrease performance to add properties on objects that are not supposed to have it.